fs/inode/fs_files.c

```diff
@@ -225,17 +235,24 @@ int files_allocate(FAR struct inode *inode, int oflags, off_t pos,
   /* The space of file array isn't enough, allocate a new filechunk */
 
   ret = files_extend(list, i + 1);
-  if (ret >= 0)
+  if (ret < 0)
```

Comment:
Why not keep `if (ret >= 0)` and add
```
  if (addref)
    {
      inode_addref(inode);
    }
```
inside?
I do not mind with current code, but we can save one `nxmutex_unlock(&list->fl_lock);` call.

Reply:
I want to move inode_addref out of lock to improve the concurrency a little bit:)